### DISCUSSION TITLE
Generating badge based on sub key in package.json

### DISCUSSION THREAD

**Discussion by therealsujitk** [badges/shields#7407] [Q&A] [answered]
To get a badge using a key in `package.json`, you can do 
```
[GET] https://img.shields.io/github/package-json/<key>/<author>/<repository>
```

Is it possible to read a sub key? For example,
```json
{
    "key": {
        "subKeyA": "value",
        "subKeyB": "value"
    }
}
```

**Comment by calebcartwright** [comment, marked as answer]
Do you have a specific use case in mind?

**Reply by therealsujitk** [reply]
Take for example from this repository's [`package.json`](https://github.com/badges/shields/blob/master/package.json) I'd like to display the value of the `email` key inside the object `bugs`.

**Reply by calebcartwright** [reply]
Thanks! Npm's manifest schema is largely fixed (I realize of course it can contain other 3rd party content, but is not generally viewed as an arbitrary source of metadata), so we do offer an assortment of badges that directly provide specific values for known keys (e.g. name, version, dependency versions, etc.) to cover common use cases, as well as the aforementioned package.json dynamic badge for grabbing top level keys. _That_ dynamic badge doesn't support json-path style queries for digging into the json, and I personally don't think it makes sense to do so given the above.

If you really want to dig lower into the package.json contents beyond what we already support, I'd suggest using our standard dynamic json badge that allows working with fully arbitrary json content but requires you to provide the url to the json.

https://img.shields.io/badge/dynamic/json?color=blue&label=foo&query=%24.bugs.email&url=https%3A%2F%2Fraw.githubusercontent.com%2Fbadges%2Fshields%2Fmaster%2Fpackage.json
![](https://img.shields.io/badge/dynamic/json?color=blue&label=foo&query=%24.bugs.email&url=https%3A%2F%2Fraw.githubusercontent.com%2Fbadges%2Fshields%2Fmaster%2Fpackage.json)

**Reply by therealsujitk** [reply]
Thanks! This works for me.
